app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,40 @@
+require 'bcrypt'
+class PasswordResetsController < ApplicationController
+  def new
+    @page_title = "Reset Password | Haxxor News"
+  end
+  
+  def create
+    @user = User.find_by(email: params[:email])
+    if @user
+      flash[:notice] = "We've sent a reset password link to the given email."
+      PasswordResetMailer.with(user: @user, token: @user.password_token).reset_email.deliver_now
+      redirect_to new_password_reset_path
+    else
+      flash[:alert] = "No account was found with the given email."
+      redirect_to new_password_reset_path
+    end
+  end
+  
+  def edit
+    @page_title = "Reset Password | Haxxor News"
+    @user = User.find_by(password_token: params[:token])
```

Comment:
You could find by both password token and their email. This may alleviate the unique index on passowrd token too.

Reply:
how would you have the email here? would that be in the link that gets included in the email?

Reply:
it's considered a best practice to timeout a sent token after a few hours. You can accomplish that by storing when a token was sent and comparing sent time to current time in the db query.

Reply:
Correct, the email would be stored into the URL that is passed to the user